@claude Please fix PR #41 CI test failure.

Good news: CodeQL now passes, and `format`, `lint`, `typecheck`, `build`, Docker, Android, and security checks pass. Only `npm test` fails.

Failing job:
https://github.com/hacksynth/ose/actions/runs/25253836279/job/74049973922

Failures are both in the new `src/test/html-to-text.test.ts`:

1. `decodes &lt; and &gt;` at line 42

```text
AssertionError: expected 'a   c' to be 'a < b > c'
Expected: "a < b > c"
Received: "a   c"
```

2. `strips residual tags in complex mixed content` at line 86

```text
AssertionError: expected result to contain '<A>'
Received:
 Q: which of  these  is right?
 [图片:img.png]
 Answer:   &
```

This looks like a mismatch between the helper's safe second-pass tag stripping and the test expectations. The implementation strips `<...>` after entity decoding, so decoded `< b >`, `<A>`, and `<B>` are removed. Please decide the intended behavior and update code/tests consistently while keeping CodeQL clean.

Constraints:
- Keep all 3 CodeQL alerts fixed.
- Preserve `<img src=...>` placeholder behavior.
- Ensure `npm test` passes.
- Keep `npm run typecheck` passing.

--- a/src/lib/utils/html-to-text.ts
+++ b/src/lib/utils/html-to-text.ts
@@ -4,6 +4,8 @@
  * Safety properties:
  *  - Two-pass tag stripping: once before and once after entity decoding,
  *    so entities like &lt;script&gt; cannot survive as tags in the output.
+ *  - Second pass uses /<[a-zA-Z\/!][^>]*>/g so only valid tag starters are
+ *    stripped; decoded text like "< b >" (space-leading) is preserved as-is.
  *  - &amp; is decoded last, preventing double-unescaping of sequences like
  *    &amp;lt; (which would otherwise become < via &amp;→& then &lt;→<).
  *  - <img> src is preserved as a [图片:URL] placeholder.
@@ -20,5 +22,5 @@ export function htmlToText(html: string): string {
     .replace(/&quot;/gi, '"')
     .replace(/&#39;/gi, "'")
     .replace(/&amp;/gi, '&')
-    .replace(/<[^>]+>/g, ' ');
+    .replace(/<[a-zA-Z\/!][^>]*>/g, ' ');
 }

--- a/src/test/html-to-text.test.ts
+++ b/src/test/html-to-text.test.ts
@@ -39,6 +39,8 @@ describe('htmlToText', () => {
   });
 
   it('decodes &lt; and &gt;', () => {
+    // Decoded "< b >" starts with a space so the second-pass tag stripper
+    // (which only matches /<[a-zA-Z\/!]...>/) leaves it intact.
     expect(htmlToText('a &lt; b &gt; c')).toBe('a < b > c');
   });
 
@@ -81,10 +83,10 @@ describe('htmlToText', () => {
     const input =
       '<p>Q: which of <b>these</b> is right?</p><img src="img.png"><br/>Answer: &lt;A&gt; &amp; &lt;B&gt;';
     const result = htmlToText(input);
+    // &lt;A&gt; and &lt;B&gt; decode to <A> and <B>; the second pass strips them
+    // because they start with a letter — consistent with stripping real HTML tags.
     expect(result).not.toMatch(/<[a-zA-Z]/);
     expect(result).toContain('[图片:img.png]');
-    expect(result).toContain('<A>');
-    expect(result).toContain('<B>');
     expect(result).toContain('&');
   });
 });

